### ISSUE TITLE
Issue with compilation: No folder named 'Build' and 'lib' present

### ISSUE THREAD

**Issue by abhimaniyar** [amoradinejad/limHaloPT#13]
I downloaded the latest version of the code which has substantial changes and started compiling it. I would like to point out three issues:

1. In the 'Dependencies' section of the README file, instructions are provided on how to change the makefile to link the CUBA and GSL libraries to the code. However, this section mentions that the makefile is in the 'Source' directory. I think that is not the case and it seems like in the latest version of the code, makefile is placed in the home directory. This very minor correction should be made in the README file.
2. After modifying the makefile to link the CUBA and GSL libraries to the code, I typed make to compile the code and I got the following error: 
```
$SOURCE_FILES = ./Source/cosmology.c ./Source/hcubature.c ./Source/ir_res.c ./Source/line_ingredients.c ./Source/main.c ./Source/ps_halo_1loop.c ./Source/ps_line_hm.c ./Source/ps_line_pt.c ./Source/read_input.c ./Source/setup_teardown.c ./Source/survey_specs.c ./Source/utilities.c ./Source/wnw_split.c
$LIB_FILES = ./Source/cosmology.c ./Source/hcubature.c ./Source/ir_res.c ./Source/line_ingredients.c ./Source/ps_halo_1loop.c ./Source/ps_line_hm.c ./Source/ps_line_pt.c ./Source/read_input.c ./Source/setup_teardown.c ./Source/survey_specs.c ./Source/utilities.c ./Source/wnw_split.c
$LIB_OBJECT_FILES = ./Build/cosmology.o ./Build/hcubature.o ./Build/ir_res.o ./Build/line_ingredients.o ./Build/ps_halo_1loop.o ./Build/ps_line_hm.o ./Build/ps_line_pt.o ./Build/read_input.o ./Build/setup_teardown.o ./Build/survey_specs.o ./Build/utilities.o ./Build/wnw_split.o
gcc -c Source/main.c -o Build/main.o -I./Include/   -I/opt/local/include -I/Class/include -I../../Cuba/Cuba-4.2.2
clang -cc1as: error: unable to open output file 'Build/main.o': 'No such file or directory'
make: *** [Build/main.o] Error 1
```
It seems like a folder named 'Build' either should be already present or should be created while compiling the code. As it is not currently present, the compilation fails. So I manually created a folder called 'Build' in the home directory and this resolved the issue but the compilation further failed which brings me to my next point.
3. After manually adding the 'Build' folder, I typed make again. Compilation was successful till I got the following error:
```
ar -rcs lib/liblimHaloPT.a ./Build/cosmology.o ./Build/hcubature.o ./Build/ir_res.o ./Build/line_ingredients.o ./Build/ps_halo_1loop.o ./Build/ps_line_hm.o ./Build/ps_line_pt.o ./Build/read_input.o ./Build/setup_teardown.o ./Build/survey_specs.o ./Build/utilities.o ./Build/wnw_split.o
ar: lib/liblimHaloPT.a: No such file or directory
make: *** [lib/liblimHaloPT.a] Error 1
```
I have only posted part of the output as the other details are not required. Similar to point no. 2, it seems like a folder called 'lib' should either be already present or created in the compilation process without which the code fails to compile.

I would suggest fixing these issue by either making arrangements to create these directories during compilation or leaving open directories called 'Build' and 'lib' in the repository.

**Comment by amoradinejad** [comment]
Dear Abhishek, 


Thank you for your feedback. 

I addressed the points you mentioned::
Corrected the “Dependencies” section to correctly state the location of makefile.
I did not realised before that since “Build” directory is an empty one (it is filled by .o files once the code is compiled), it is excluded from git push. I placed a .gitkeep file inside the directory  so that now the Build directory is included in git push/pull.
Similar to Build folder, I created a .gitkeep file in “lib” directory and excluded “liblimHaloPT.a” file from git updates. The static library .a file is generated and placed in this folder upon compilation of the package by the user.

Best,
Azadeh
